demo/core_13_$.utils.code.js

```diff
@@ -136,3 +136,60 @@ $.utils.code.toSourceSafe = function(value) {
     throw e;
   }
 };
+
+$.utils.code.formatForEval = function(src) {
+  // Eval treats {} as an empty block (return value undefined).
+  // Eval treats {'a': 1} as a syntax error.
+  // Eval treats {a: 1} as block with a labeled statement (return value 1).
+  // Detect these cases and enclose in parenthesis.
+  // But don't mess with: {var x = 1; x + x;}
+  // This is consistent with the console on Chrome and Node.
+  try {
+    var ast = $.utils.acorn.parse(src, { ecmaVersion: 5 });
+  } catch (e) {
+    try {
+      ast = $.utils.acorn.parseExpressionAt(src, 0, { ecmaVersion: 5 });
+      if (src.substring(ast.end).trim() !== '') {
+        throw SyntaxError('Not just an expression');
+      }
+    } catch (e) {
+      // This eval will fail.  Have fun.
+      return src;
+    }
+    // This is an object literal: {'a': 1}
+    return '(' + src + ')';
+  }
+  if (ast.type === 'Program' && ast.body.length === 1 &&
+      ast.body[0].type === 'BlockStatement') {
+    if (ast.body[0].body.length === 0) {
+      // This is an empty object: {}
+      return '(' + src + ')';
+    }
+    if (ast.body[0].body.length === 1 &&
+        ast.body[0].body[0].type === 'LabeledStatement' &&
+        ast.body[0].body[0].body.type === 'ExpressionStatement') {
+      // This is an object literal: {a: 1}
+      return '(' + src + ')';
+    }
+  }
+  return src;
+};
+
+$.utils.code.formatForEval.test = function() {
```

Comment:
Maybe `__test__` or something a bit more distinctive?  After all, there is already a `RegExp.prototype.test` which is not unit tests for `RegExp.prototype`.

Reply:
How about `unittest`?  I'm not really a fan of adding more and more underscores for uniqueness.

Reply:
Ahh.  Whereas I would prefer to avoid a name that does not have underscores, least it conflict with some purpose other than unit testing.  I am specifically thinking along the lines of having unit tests run automatically—a situation in which you do not want random, similarly-named functions being called.

(The correct solution would be a unique Symbol, of course…)